### DISCUSSION TITLE
Ctrlr doesn't receive (or send) data to Line6 POD XT Live on Linux

### DISCUSSION THREAD

**Discussion by oli747** [RomanKubiak/ctrlr#480] [General]
Hi all,
many thanks for ctrlr, I previously successfully created a panel to edit the patches for a Roland GP-8 Guitar multi effects processor, working like a charm! 

Today I wanted to start on a panel to edit a Line6 POD XT Live (a guitar amp modeller). It supports midi over USB, which is supported on Linux. Ctrlr detects the midi device correctly (correctly named POD XT Live), but it doesn't seem to send or detect any midi data. Using midisnoop I see the midi control data when I turn any of the knobs on the device, so the Linux driver is working as expected. But on the midi monitor of Ctrlr I don't see any data when I turn that same knob. Sending a program change also doesn't give the desired effect on the POD XT Live. I checked the channel (and also tried omni mode), but no results. How can I debug this issue? What is the difference between the way midisnoop and Ctrlr read midi data? Any hints what I should look for?

**Comment by dnaldoog** [comment]
@oli747 

Which version of Ctrlr are you using?

**Reply by oli747** [reply]
I tried several versions on a different computer with Ubuntu 20.04:
Ctrl-6.0.35 crashes with a segfault
Ctrl-5.5.9 crashes with a segfault
Ctrl-5.3.201 works !!!!!

then I went back to the previous computer. It seems that this had an older version, I installed 5.3.201 and it worked too. Do you need any information on the crashes? (and how can I provide that information?)

**Comment by dnaldoog** [comment]
I don't think the crash data would make much sense to most people, but I was going to say __5.3.201__ is the best version for linux and windows. Looks like you've discovered that.

What is the USB to MIDI device? If it's one of those cheap Chinese ones, then throw it away. Can you connect MIDI DIN directly through an audio interface or something like that?

This program can receive MIDI data no problem. Maybe check to see if it works using this?
https://sourceforge.net/projects/ctrlr-utilities/files/GSDR.bpanelz/download

I use Debian 11.0 with Ctrlr 5.3.201 and panels I load seem to work, although I don't use linux for audio things extensively.

Finally, can you dual boot into Windows and check MIDI connectivity using the same hardware?

**Reply by oli747** [reply]
I know how to use gdb so if I can help you with providing backtrace information then just ask.

I don't connect to the midi ports - the device has a USB port, and has an Alsa driver on Linux that supports MIDI and digital audio. 

I'm not sure the problem is completely resolved - yesterday I experienced it again with 5.3.201. I'll dive into that this evening (sorry no windows around here, so I cannot dual boot).
